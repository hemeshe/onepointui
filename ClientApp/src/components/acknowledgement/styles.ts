import styled from 'styled-components';

import { Row } from '../grid';

type InstructionProps = {
  borderRight?: boolean;
  borderLeft?: boolean;
};

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
`;

export const StyledRow = styled(Row)`
  padding: 10px;
  overflow-wrap: anywhere;
`;

export const Button = styled.button<{ backgroundColor?: string }>`
  outline: none;
  border: none;
  background: ${({ backgroundColor }) =>
    backgroundColor ? backgroundColor : '#e6e6e6'};
  border-radius: 4px;
  color: black;
  letter-spacing: 1px;
  padding: 6px 12px;
  font-size: ${({ theme }) => theme.fontSize['xs']};
  cursor: pointer;
  margin: 2px 8px;
`;
