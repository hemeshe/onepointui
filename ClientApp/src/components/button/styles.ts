import styled, { css } from 'styled-components/macro';

import { Props } from './Button';

export const StyledButton = styled.button<Props>`
  -webkit-appearance: button;
  display: inline-block;
  font-weight: 400;
  line-height: 1.5;
  color: #003c88;
  text-align: center;
  text-decoration: none;
  vertical-align: middle;
  cursor: ${({ disabled }) => (disabled ? 'not-allowed' : 'pointer')};
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: transparent;
  border: 1px solid transparent;
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  border-radius: 0.25rem;
  transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out,
    border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;

  ${({ primary, theme, size }) => {
    if (primary) {
      return css`
        background-color: ${theme.buttonColors['primaryBackground']};
        color: ${theme.buttonColors['primaryFont']};
        border-color: ${theme.buttonColors['primaryBorder']};
      `;
    }
    if (size === 'block') {
      return css`
        display: block;
        width: 100%;
      `;
    }
  }}

  ${({ size }) => {
    if (size === 'block') {
      return css`
        display: block;
        width: 100%;
      `;
    }
  }}
`;
