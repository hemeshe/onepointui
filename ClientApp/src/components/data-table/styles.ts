import styled from "styled-components";

export const SortButton = styled.span`
  cursor: pointer;
  margin-left: 10px;
`;

export const Input = styled.input`
  background: #e6e6e6;
  border: 1px solid #c2c2c2;
  padding: 5px;
  border-radius: 4px;
  display: block;
  max-width: 100px;
`;

export const Checkbox = styled.input`
  display: block;
`;
