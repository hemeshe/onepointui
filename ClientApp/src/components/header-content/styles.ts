import styled from 'styled-components';
import { Link } from 'react-router-dom';

type NavItemProps = {
  activelink?: boolean;
};

export const HeaderContent = styled.section`
  display: flex;
  width: 100%;
  justify-content: space-between;
  height: ${({ theme }) => theme.headerHeight};
`;

export const Left = styled.div`
  display: flex;
  align-items: center;
  width: 27%;
`;

export const Logo = styled.div``;

export const TitleContainer = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  font-family: Futura Medium;
  color: #003c88;
  justify-content: center;
  margin: 0 25px;
  font-size: ${({ theme }) => theme.fontSize['l']};
`;

export const SubTitleContainer = styled.div`
  display: flex;
  justify-content: flex-start;
  width: 100%;
  font-weight: normal;
  font-size: ${({ theme }) => theme.fontSize['s']};
  letter-spacing: 1px;
`;

export const SubTitle = styled.div`
  padding: 0 4px;
`;
export const Title = styled.div`
  font-weight: 900;
  word-spacing: ${({ theme }) => theme.spacing['xxxs']};
  letter-spacing: ${({ theme }) => theme.spacing['xxxs']};
`;

export const SubTitleLabel = styled.div`
  display: flex;
  height: 100%;
  align-items: center;
  justify-content: center;
`;

export const Center = styled.div`
  display: flex;
  width: 60%;
  background: none;
  align-items: center;
  height: 100%;
`;

export const NavMain = styled.div`
  display: flex;
  justify-content: space-between;
  height: 100%;
  align-items: center;
`;

export const NavItem = styled.span<NavItemProps>`
  height: 100%;
  border-bottom: ${({ activelink }) =>
    activelink ? '3px solid #009EB4' : 'none'};
  display: flex;
  align-items: center;
  box-sizing: border-box;
  margin: 14px 20px;
  font-weight: ${({ activelink }) => (activelink ? 'bold' : 'normal')};
`;

export const NavItemLink = styled(Link)`
  text-decoration: none;
  color: #003c88;
  font-size: ${({ theme }) => theme.fontSize['m']};
`;

export const Right = styled.div`
  display: flex;
  align-items: center;
  width: 20%;
  justify-content: flex-end;
`;

export const NavIconItem = styled.div`
  display: inline-block;
  position: relative;
`;
export const NavIconLink = styled(Link)`
  text-decoration: none;
  margin: 4px 8px;
  color: #003c88;
  position: relative;
`;

export const UserIcon = styled.div`
  display: flex;
  border-radius: 50px;
  width: 36px;
  height: 36px;
  background: #ededed;
  color: #3363a0;
  justify-content: center;
  align-items: center;
  letter-spacing: ${({ theme }) => theme.spacing['xxxs']};
  font-size: ${({ theme }) => theme.fontSize['m']};
`;
