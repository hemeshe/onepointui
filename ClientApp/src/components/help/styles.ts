import styled from "styled-components";

export const Help = styled.div`
  width: 100%;
  display: flex;

  flex-direction: column;
  position: absolute;
  top: 0;
`;

export const HelpCard = styled.div`
  display: flex;
  z-index: 1;
  flex-direction: column;
  position: absolute;
  top: 0;
  left: -470px;
  width: 470px;
  height: auto;
  box-sizing: content-box;
  background: #ffffff;
  padding: 10px 6px 0px 6px;
  border: 1px solid #ededed;
  box-sizing: border-box;
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.35);
`;

export const Tip = styled.p`
  width: 100%;
  background: #ffffff;
`;

export const HelpTitle = styled.span`
  font-family: Futura Medium;
  font-style: normal;
  font-weight: bold !important;
  font-size: 1.2rem;
  /* identical to box height */
  text-align: center;
  display: block;
  /* dark-blue */

  color: #003c88;
  margin-bottom: 2px;
`;

export const HeplContent = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
  color: #003c88;
`;

export const HelpDesc = styled.span`
  display: block;
  padding: 0 2px 18px 2px;
  text-align: center;
  font-size: 1rem;
`;

export const StyledEmail = styled.a`
  margin: 0;
  padding: 0;
  text-decoration: none;
  color: #641964;
`;
