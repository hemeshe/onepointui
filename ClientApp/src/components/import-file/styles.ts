import styled from "styled-components";

import { Col } from "../grid";

type InstructionProps = {
  borderRight?: boolean;
  borderLeft?: boolean;
};

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
`;

export const Main = styled.div`
  border: 2px dashed #595959;
  border-radius: 4px;
  width: 100%;
  padding: 15px;
  display: flex;
  flex-direction: column;
`;

export const Instruction = styled.div<InstructionProps>`
  display: block;
  color: #000000;
  font-size: ${({ theme }) => theme.fontSize["xs"]};
  border-right: ${({ borderRight }) =>
    borderRight ? "1px solid #7F7F7F" : ""};
  border-left: ${({ borderLeft }) => (borderLeft ? "1px solid #7F7F7F" : "")};
  padding: 10px;
`;

export const LinkInput = styled.input`
  display: block;
  width: 100%;
  padding: 10px;
  margin: 10px;
`;

export const Footer = styled.div`
  display: block;
  padding: 10px;
`;

export const Button = styled.button`
  background: #a6a6a6;
  border: 1px solid #a6a6a6;
  border-radius: 4px;
  padding: 10px 25px;
  cursor: pointer;
  color: white;
`;

export const FileDragArea = styled(Col)`
  position: relative;
  width: 100%;

  padding: 20px;
  display: flex;
  flex-direction: column;
  cursor: pointer;
`;

export const FileDragDesc = styled.span`
  display: block;
  padding: 10px;
  text-align: center;
  font-size: ${({ theme }) => theme.fontSize["m"]};
`;

export const Browse = styled.label`
  color: #89cfdc;
`;

export const FileInput = styled.input`
  position: absolute;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  outline: none;
  opacity: 0;
  border: 0;
  outline: none;
  ::-webkit-file-upload-button {
    visibility: hidden;
  }
`;

export const FileOutputArea = styled(Col)`
  position: relative;
  border: 1px solid #a6a6a6;
  box-sizing: border-box;
  border-radius: 4px;
  padding: 5px;
`;

export const FileName = styled(Col)``;

export const FileRemove = styled(Col)`
  padding: 0 5px;
  cursor: pointer;
`;

export const ErrorMessage = styled(Col)`
  color: #e72d30;
`;
