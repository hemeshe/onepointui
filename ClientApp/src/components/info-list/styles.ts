import styled from "styled-components";

import { Row, Col } from "../grid";

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
`;

export const InfoRow = styled(Row)<{ backgroundColor?: string }>`
  color: #003c88;
  padding: 0 20px !important;
  box-sizing: border-box;
  min-width: 100%;
  margin: 0 !important;
  cursor: pointer;
  background-color: ${({ backgroundColor }) =>
    backgroundColor ? backgroundColor : "inherit"};

  :hover {
    background-color: #89cfdc;
  }
`;

export const InfoCol = styled(Col)<{ borderBottom?: boolean }>`
  padding: 11px 10px;
  border-bottom: ${({ borderBottom }) =>
    borderBottom ? "1px solid #D9D9D9" : ""};
  box-sizing: border-box;

  :hover {
    border: none;
  }
`;

export const Text = styled.span`
  padding: 0 10px 0 0;
  font-size: 1rem;
`;

export const NavImg = styled.img`
  float: right;
`;
