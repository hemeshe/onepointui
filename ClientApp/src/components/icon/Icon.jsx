import React from "react";
import styled, { css } from "styled-components";

import { ReactComponent as Logo } from "./icons/logo.svg";
import { ReactComponent as Home } from "./icons/home.svg";
import { ReactComponent as EvpCentral } from "./icons/evp-central.svg";
import { ReactComponent as Operations } from "./icons/operations.svg";
import { ReactComponent as Shipping } from "./icons/shipping.svg";
import { ReactComponent as ShellEnergy } from "./icons/shell-energy.svg";
import { ReactComponent as Crude } from "./icons/crude.svg";
import { ReactComponent as Products } from "./icons/products.svg";
import { ReactComponent as Help } from "./icons/help.svg";
import { ReactComponent as ScoreCard } from "./icons/score-card.svg";
import { ReactComponent as PerformanceTracker } from "./icons/performance-tracker.svg";
import { ReactComponent as OverallTracker } from "./icons/overall-tracker.svg";
import { ReactComponent as Cashflow } from "./icons/cashflow.svg";
import { ReactComponent as WorkingCapital } from "./icons/working-capital.svg";
import { ReactComponent as TradingAllocations } from "./icons/trading-allocations.svg";
import { ReactComponent as CostReporting } from "./icons/cost-reporting.svg";
import { ReactComponent as CapitalEmployed } from "./icons/capital-employed.svg";
import { ReactComponent as DefaultIcon } from "./icons/default-icon.svg";
import { ReactComponent as ReportIcon } from "./icons/report-icon.svg";
import { ReactComponent as SmallHelpIcon } from "./icons/small-help.svg";
import { ReactComponent as InfoIcon } from "./icons/info.svg";
import { ReactComponent as UserAvatarIcon } from "./icons/user-avatar.svg";
import { ReactComponent as MinusIcon } from "./icons/Minus.svg";
import { ReactComponent as PlusIcon } from "./icons/plus.svg";
import { ReactComponent as CaretIcon } from "./icons/caret.svg";
import { ReactComponent as CaretIconDown } from "./icons/caret-down.svg";
import { ReactComponent as UploadIcon } from "./icons/upload.svg";
import { ReactComponent as RightCaretIcon } from "./icons/right-caret.svg";
import { ReactComponent as CloseIcon } from "./icons/close.svg";

export const icons = {
  logo: Logo,
  home: Home,
  evpCentral: EvpCentral,
  operations: Operations,
  shipping: Shipping,
  shellEnergy: ShellEnergy,
  crude: Crude,
  products: Products,
  help: Help,
  scoreCard: ScoreCard,
  performanceTracker: PerformanceTracker,
  overallTracker: OverallTracker,
  cashflow: Cashflow,
  workingCapital: WorkingCapital,
  tradingAllocations: TradingAllocations,
  costReporting: CostReporting,
  capitalEmployed: CapitalEmployed,
  defaultIcon: DefaultIcon,
  reportIcon: ReportIcon,
  smallHelpIcon: SmallHelpIcon,
  infoIcon: InfoIcon,
  userAvatarIcon: UserAvatarIcon,
  minusIcon: MinusIcon,
  plusIcon: PlusIcon,
  caretIcon: CaretIcon,
  uploadIcon: UploadIcon,
  rightCaretIcon: RightCaretIcon,
  closeIcon: CloseIcon,
  caretDownIcon: CaretIconDown,
};

const IconBase = ({ type, ...rest }) => {
  const Component = icons[type];
  return <Component {...rest} />;
};

export const Icon = styled(IconBase)`
  ${({ theme, color, height, width, size }) => {
    const heightSpacing = height || size;
    const widthSpacing = width || size;

    return css`
      height: ${heightSpacing ? theme.size[heightSpacing] : "100%"};
      width: ${widthSpacing ? theme.size[widthSpacing] : "100%"};
      fill: ${color ? color : "currentColor"};

      @media (max-width: 1280px) {
        max-height: 56px;
      }
    `;
  }}
`;
