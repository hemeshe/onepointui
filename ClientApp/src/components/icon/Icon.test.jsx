import React from "react";
import { renderWithTheme } from "../../styles/render-with-theme";

import { Icon } from "./Icon.jsx";

describe("Icon", () => {
  it("should render with the default height & width", () => {
    const { container } = renderWithTheme(<Icon type="logo" />);
    expect(container).toMatchSnapshot();
  });
  it("should render with the given width and height", () => {
    const { container } = renderWithTheme(
      <Icon type="logo" width={24} height={24} />
    );
    expect(container).toMatchSnapshot();
  });
  it("should render with the given size", () => {
    const { container } = renderWithTheme(<Icon type="logo" size={24} />);
    expect(container).toMatchSnapshot();
  });
});
