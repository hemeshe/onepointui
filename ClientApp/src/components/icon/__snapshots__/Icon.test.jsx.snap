// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should render with the default height & width 1`] = `
.c0 {
  height: 100%;
  width: 100%;
  fill: currentColor;
}

@media (max-width:1280px) {
  .c0 {
    max-height: 56px;
  }
}

<div>
  <svg
    class="c0"
  >
    logo.svg
  </svg>
</div>
`;

exports[`Icon should render with the given size 1`] = `
.c0 {
  height: 24px;
  width: 24px;
  fill: currentColor;
}

@media (max-width:1280px) {
  .c0 {
    max-height: 56px;
  }
}

<div>
  <svg
    class="c0"
    size="24"
  >
    logo.svg
  </svg>
</div>
`;

exports[`Icon should render with the given width and height 1`] = `
.c0 {
  height: 24px;
  width: 24px;
  fill: currentColor;
}

@media (max-width:1280px) {
  .c0 {
    max-height: 56px;
  }
}

<div>
  <svg
    class="c0"
    height="24"
    width="24"
  >
    logo.svg
  </svg>
</div>
`;
